src/kvstore.c

```diff
@@ -568,14 +577,23 @@ kvstoreIterator *kvstoreIteratorInit(kvstore *kvs) {
 void kvstoreIteratorRelease(kvstoreIterator *kvs_it) {
     dictIterator *iter = &kvs_it->di;
     dictResetIterator(iter);
-
+    freeDictIfNeeded(kvs_it->kvs, kvs_it->didx);
```

Comment:
so these are needed because safe iterators call dictPauseRehashing, and now that we have the check in freeDictIfNeeded, we have to repeat that when releasing the iterator, right?

let's add a comment (in each of the new calls to freeDictIfNeeded).

Reply:
i added the new comment in freeDictIfNeeded